firmware/inc/rtc.h

```diff
@@ -5,38 +5,56 @@
 
 #include "clock_types.h"
 
+#define MAX_NO_DAYS_RESET   5
+
+typedef enum ALARM_STATUS{
+    ALARM_DISABLE,
+    ALARM_ENABLE
+}ALARM_STATUS;
+
+typedef enum ALARM_ID{
+    ALARM = 1,
+    TIMER = 2
+}ALARM_ID;
+
 typedef struct rtc_time_t
 {
     uint8_t hr;
     uint8_t min;
     uint8_t sec;
-    uint8_t date;
-    uint8_t month;
-    uint8_t dow;
-    uint16_t year;
 } RtcTime;
 
 typedef struct rtc_t
 {
     void (*setRtcTime)(
         uint8_t hour_24mode,
         uint8_t minute,
-        uint8_t second,
-        uint8_t date,
-        uint8_t month,
-        uint8_t dow,
-        uint16_t year);
-    uint8_t (*getDayOfWeek)(void);
-    uint8_t (*getDate)(void);
-    uint8_t (*getMonth)(void);
-    uint16_t (*getYear)(void);
+        uint8_t second);
+    void (*setDay)(uint8_t day);
+    void (*setMonth)(uint8_t month);
+    void (*enableAlarm)(ALARM_ID id, ALARM_STATUS enable);

```

Comment:
We can't use types only defined in this module as parameters for function pointers from other modules. These will be populated with functions from i2c-rtc.h and the enableAlarm function takes a uint8_t and bool as inputs. Types have to match these functions